mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java

```diff
@@ -527,6 +528,20 @@ public void checkParamsAndMetrics() {
     assertTag(tags, "user_email", USER_EMAIL);
   }
 
+  @Test
+  public void getMetricHistoryPagination() {
+    String expId = client.createExperiment("getMetricHistoryPagination");
+    RunInfo run = client.createRun(expId);
+    String runId = run.getRunUuid();
+    List<Long> steps = LongStream.range(0, 26000).boxed().collect(Collectors.toList());
+    for (Long step: steps) {
+      client.logMetric(runId, "random_metric", step, step, step);
+    }
+    List<Metric> metrics = client.getMetricHistory(runId, "random_metric");
+    List<Double> values = steps.stream().mapToDouble(x -> x).boxed().collect(Collectors.toList());
```

Comment:
nice job here with avoiding the additional overhead of auto boxing the cast to Double.